great_expectations/self_check/util.py

```diff
@@ -1513,9 +1518,11 @@ def build_test_backends_list(  # noqa: C901, PLR0912, PLR0913, PLR0915 # FIXME C
         if include_redshift:
             # noinspection PyUnresolvedReferences
             try:
-                engine = _create_redshift_engine()
-                conn = engine.connect()
-                conn.close()
+                if "redshift" not in _successful_backend_connection:
+                    engine = _create_redshift_engine()
+                    conn = engine.connect()
+                    conn.close()
+                    _successful_backend_connection.add("redshift")
```

Comment:
I thought this only applied to legacy testing, and I don't see any changes to legacy tests. Does this code get run in production? I'm just now realizing that I don't understand why this entire file isn't in the `tests` directory.

Reply:
I think this file should be in the tests directory. Historically I think we had some tools that let people check their yaml strings that lived in this file. After talking with product today, I pulled in a ticket to delete some legacy tests and hopefully that will allow be to delete this file though I haven't tracked down everywhere this file is used yet.

Reply:
Oh in terms of why this is here. redshift is in the backend list so it now tries to connect everytime this function is called. It takes seconds to connect to redshift and this function actually gets called multiple times during tests.